src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/VariantRecalibrator.java

```diff
@@ -0,0 +1,852 @@
+     * preferable to provide this tool with more data (typically by including more samples or more territory)
+     * in order to generate a more robust model.
+     */
+    @Advanced
+    @Argument(fullName="max_attempts",
+            shortName = "max_attempts",
+            doc="Number of attempts to build a model before failing",
+            optional=true)
+    private int max_attempts = 1;
+
+    /////////////////////////////
+    // Debug Arguments
+    /////////////////////////////
+    @Advanced
+    @Argument(fullName = "trustAllPolymorphic",
+            shortName = "allPoly",
+            doc = "Trust that all the input training sets' unfiltered records contain only polymorphic sites to drastically speed up the computation.",
+            optional=true)
+    private Boolean TRUST_ALL_POLYMORPHIC = false;
+
+    /////////////////////////////
+    // Private Member Variables
+    /////////////////////////////
+    private VariantDataManager dataManager;
+    private VariantContextWriter recalWriter;
+    private PrintStream tranchesStream;
+    private ArrayList<Double> replicate = new ArrayList<>();
+    private final Set<String> ignoreInputFilterSet = new TreeSet<>();
+    private final VariantRecalibratorEngine engine = new VariantRecalibratorEngine( VRAC );
+    private ExpandingArrayList<VariantDatum> reduceSum = new ExpandingArrayList<>();
+
+    //---------------------------------------------------------------------------------------------------------------
+    //
+    // initialize
+    //
+    //---------------------------------------------------------------------------------------------------------------
+
+    @Override
+    public void onTraversalStart() {
+        dataManager = new VariantDataManager( new ArrayList<>(USE_ANNOTATIONS), VRAC );
+
+        if (RSCRIPT_FILE != null && !RScriptExecutor.RSCRIPT_EXISTS)
+            Utils.warnUser(logger, String.format(
+                    "Rscript not found in environment path. %s will be generated but PDF plots will not.",
+                    RSCRIPT_FILE));
+
+        if( IGNORE_INPUT_FILTERS != null ) {
+            ignoreInputFilterSet.addAll( IGNORE_INPUT_FILTERS );
+        }
+
+        try {
+            tranchesStream = new PrintStream(TRANCHES_FILE);
+        } catch (FileNotFoundException e) {
+            throw new UserException.CouldNotCreateOutputFile(TRANCHES_FILE, e);
+        }
+
+        for( FeatureInput<VariantContext> variantFeature : resource ) {
+            dataManager.addTrainingSet( new TrainingSet( variantFeature ) );
+        }
+
+        if( !dataManager.checkHasTrainingSet() ) {
+            throw new UserException.CommandLineException(
+                    "No training set found! Please provide sets of known polymorphic loci marked with the training=true ROD binding tag. For example, -resource:hapmap,VCF,known=false,training=true,truth=true,prior=12.0 hapmapFile.vcf" );
```

Comment:
Remove mention of "ROD bindings" from these error messages, replacing with GATK4 terminology.